perception/autoware_tensorrt_bevdet/src/bevdet.cpp

```diff
@@ -0,0 +1,717 @@
+
+  CHECK_CUDA(cudaMemcpy(
+    trt_buffer_dev_[buffer_map_["mean"]], mean_.data(), 3 * sizeof(float), cudaMemcpyHostToDevice));
+  CHECK_CUDA(cudaMemcpy(
+    trt_buffer_dev_[buffer_map_["std"]], std_.data(), 3 * sizeof(float), cudaMemcpyHostToDevice));
+}
+
+void BEVDet::initCamParams(
+  const std::vector<Eigen::Quaternion<float>> & curr_cams2ego_rot,
+  const std::vector<Eigen::Translation3f> & curr_cams2ego_trans,
+  const std::vector<Eigen::Matrix3f> & curr_cams_intrin)
+{
+  for (int i = 0; i < n_img_; i++) {
+    cam_params_host_[i * cam_params_size_ + 0] = curr_cams_intrin[i](0, 0);
+    cam_params_host_[i * cam_params_size_ + 1] = curr_cams_intrin[i](1, 1);
+    cam_params_host_[i * cam_params_size_ + 2] = curr_cams_intrin[i](0, 2);
+    cam_params_host_[i * cam_params_size_ + 3] = curr_cams_intrin[i](1, 2);
+    cam_params_host_[i * cam_params_size_ + 4] = post_rot_(0, 0);
+    cam_params_host_[i * cam_params_size_ + 5] = post_rot_(0, 1);
+    cam_params_host_[i * cam_params_size_ + 6] = post_trans_.translation()(0);
+    cam_params_host_[i * cam_params_size_ + 7] = post_rot_(1, 0);
+    cam_params_host_[i * cam_params_size_ + 8] = post_rot_(1, 1);
+    cam_params_host_[i * cam_params_size_ + 9] = post_trans_.translation()(1);
+    cam_params_host_[i * cam_params_size_ + 10] = 1.f;  // bda 0 0
+    cam_params_host_[i * cam_params_size_ + 11] = 0.f;  // bda 0 1
+    cam_params_host_[i * cam_params_size_ + 12] = 0.f;  // bda 1 0
+    cam_params_host_[i * cam_params_size_ + 13] = 1.f;  // bda 1 1
+    cam_params_host_[i * cam_params_size_ + 14] = 1.f;  // bda 2 2
+    cam_params_host_[i * cam_params_size_ + 15] = curr_cams2ego_rot[i].matrix()(0, 0);
+    cam_params_host_[i * cam_params_size_ + 16] = curr_cams2ego_rot[i].matrix()(0, 1);
+    cam_params_host_[i * cam_params_size_ + 17] = curr_cams2ego_rot[i].matrix()(0, 2);
+    cam_params_host_[i * cam_params_size_ + 18] = curr_cams2ego_trans[i].translation()(0);
+    cam_params_host_[i * cam_params_size_ + 19] = curr_cams2ego_rot[i].matrix()(1, 0);
+    cam_params_host_[i * cam_params_size_ + 20] = curr_cams2ego_rot[i].matrix()(1, 1);
+    cam_params_host_[i * cam_params_size_ + 21] = curr_cams2ego_rot[i].matrix()(1, 2);
+    cam_params_host_[i * cam_params_size_ + 22] = curr_cams2ego_trans[i].translation()(1);
+    cam_params_host_[i * cam_params_size_ + 23] = curr_cams2ego_rot[i].matrix()(2, 0);
+    cam_params_host_[i * cam_params_size_ + 24] = curr_cams2ego_rot[i].matrix()(2, 1);
+    cam_params_host_[i * cam_params_size_ + 25] = curr_cams2ego_rot[i].matrix()(2, 2);
+    cam_params_host_[i * cam_params_size_ + 26] = curr_cams2ego_trans[i].translation()(2);
+  }
+  CHECK_CUDA(cudaMemcpy(
+    trt_buffer_dev_[buffer_map_["cam_params"]], cam_params_host_,
+    trt_buffer_sizes_[buffer_map_["cam_params"]], cudaMemcpyHostToDevice));
+}
+
+void BEVDet::initParams(const std::string & config_file)
+{
+  mean_ = std::vector<float>(3);
```

Comment:
Consider making `mean_` and `std_` private members of the `BEVDet` class and providing `const` accessors to adhere to better encapsulation principles. This improves data hiding and reduces the chance of unintended modification.